### PR TITLE
Assume role and MFA support for Serverless CLI

### PR DESCRIPTION
**Pull request by kennu**
## What did you implement:

With these changes Serverless Framework should be compatible with
AWS CLI, so that these instructions for assuming a role work
out-of-the-box:
https://docs.aws.amazon.com/cli/latest/userguide/cli-roles.html

Related to #5048

## How did you implement it:

This sets the AWS_SDK_LOAD_CONFIG environment variable which allows
AWS SDK to read ~/.aws/config, and implements the tokenCodeFn
callback which asks the user to enter a MFA code. Credentials are
also cached to avoid creating the object multiple times, which
makes MFA work properly when multiple requests are made.

## How can we verify it:

Deploy any project using an AWS profile that has role_arn, source_profile and mfa_serial configured in ~/.aws/config.

## Todos:

- [ ] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [ ] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/aws/provider/awsProvider.js [hunk 1]
```diff
@@ -13,13 +13,17 @@ const fs = require('fs');
...
+const readline = require('readline');
 
 const constants = {
   providerName: 'aws',
 };
 
 PromiseQueue.configure(BbPromise.Promise);
 
+// Store credentials in this variable to avoid creating them several times (messes up MFA).
+let cachedCredentials = null;
```
Comment (dschep) [inline lib/plugins/aws/provider/awsProvider.js]: Move to `this.cachedCredentials` in the `AwsProvider`'s constructor.

Comment (kennu) [reply lib/plugins/aws/provider/awsProvider.js]: I think [x] Allow edits from maintainers is enabled, should I enable something more to make it work?

Comment (dschep) [reply lib/plugins/aws/provider/awsProvider.js]: Doh. You're right, I'm just not used to pushing to other peoples repos. Got it now.

**Comment by kennu** [comment]
I guess the tests are not compatible with environment variable AWS_SDK_LOAD_CONFIG=true. Should I just remove it and have users manually set it? I was hoping to eliminate that step, but I guess it could break backwards compatibility.

**Comment by dschep** [comment]
I think this solution is probably for the best anyway, don't want to set too many env vars around AWS without the user's input on it.

**Comment by kennu** [comment]
I guess this still needs some kind of test for the interactive MFA query..

**Comment by dschep** [comment]
Yeah, It'd be preferable if that function had coverage.

**Comment by kennu** [comment]
Thanks for the merge. I've been trying to figure a way to test readline() (stdin / stdout) and AWS MFA, but it's actually pretty complicated in any meaningful way.

**Pull request merged**